sklearn/pipeline.py

```diff
@@ -964,14 +967,26 @@ def _validate_transformer_weights(self):
                     "but it is not present in transformer_list."
                 )
 
+    def _passthrough_function(self):
+        ft = FunctionTransformer(accept_sparse=True, check_inverse=False)
+        ft.get_feature_names = lambda: []
+        return ft
+
     def _iter(self):
         """
         Generate (name, trans, weight) tuples excluding None and
         'drop' transformers.
         """
         get_weight = (self.transformer_weights or {}).get
+
         return (
-            (name, trans, get_weight(name))
+            (
+                name,
+                self._passthrough_function(),
```

Comment:
Yeah, I added it directly initially. I misinterpreted the function of `get_feature_names` and assumed that it should return a value for all transforms. But since it is expected to return error if the `get_feature_names` is not implemented by the transformer, I understand that it is not necessary to implement the `get_feature_names` explicitly.